pkg/engine/params_k8s.go

```diff
@@ -27,8 +34,16 @@ func assignKubernetesParameters(properties *api.Properties, parametersMap params
 		k8sVersion := orchestratorProfile.OrchestratorVersion
 		k8sComponents := api.K8sComponentsByVersionMap[k8sVersion]
 		kubernetesConfig := orchestratorProfile.KubernetesConfig
+
 		kubernetesImageBase := kubernetesConfig.KubernetesImageBase
 
+		// This is a temporary work-around as we migrate images over to mcr.
+		hyperkubeImageBase := kubernetesImageBase
+		if kubernetesImageBase == "" {
```

Comment:
Just to make sure I understand, this will be empty for public cloud and non-empty for sovereign clouds + custom image base passed in?

Reply:
That's the general idea, yes, but...
Where are sovereign cloud defaults defined?

Reply:
Yes, see https://github.com/Azure/aks-engine/pull/753/files#diff-f41c9478d3eabb30a6e28202cbfe7e18R233 for an example

Reply:
Ok so yes.
The idea here is a user spec could very well define k8s.gcr.io and I don't want to overwrite that value with our own.
But if the user config does not specify a value we can go ahead and insert in our own value for the images that are currently available on mcr.

Reply:
There may be a better place to define this logic, though.

Reply:
I'm not sure I'm seeing a better place taking another glance at it.

Reply:
do we also want to use MCR in any sovereign clouds? Or is it meant to be public cloud only?

Reply:
I'm not sure relying on `kubernetesImageBase` to be empty is a very reliable way to do this

Reply:
In any case, we definitely need unit tests for this